csrc/ir/nodes.cpp

```diff
@@ -4336,4 +4336,135 @@ std::vector<PolymorphicValue> LinearOp::evaluate(
   return {at::linear(a, b)};
 }
 
+SdpaFwdOp::SdpaFwdOp(
+    IrBuilderPasskey passkey,
+    TensorView* output,
+    TensorView* log_sumexp,
+    TensorView* cum_seq_q,
+    TensorView* cum_seq_k,
+    Val* query_seq_len,
+    Val* key_seq_len,
+    TensorView* philox_seed,
+    TensorView* philox_offset,
+    TensorView* debug_attn_mask,
+    Val* query,
+    Val* key,
+    Val* value,
+    Val* dropout_p,
+    Val* is_causal,
+    Val* scale)
+    : Expr(passkey) {
+  addOutput(output);
+  addOutput(log_sumexp);
+  addOutput(cum_seq_q);
+  addOutput(cum_seq_k);
+  addOutput(query_seq_len);
+  addOutput(key_seq_len);
+  addOutput(philox_seed);
+  addOutput(philox_offset);
+  addOutput(debug_attn_mask);
+
+  addInput(query);
+  addInput(key);
+  addInput(value);
+  addInput(dropout_p);
+  addInput(is_causal);
+  if (scale != nullptr) {
+    addInput(scale);
+  }
+}
+
+NVFUSER_DEFINE_CLONE_AND_CREATE(SdpaFwdOp)
+
+std::string SdpaFwdOp::toString(int indent_size) const {
+  std::stringstream ss;
+  indent(ss, indent_size) << attn_out()->toString() << "\n";
+  indent(ss, indent_size + 1) << " = sdpa(" << query()->toString() << ",\n";
+  indent(ss, indent_size + 1) << "          " << key()->toString() << ",\n";
+  indent(ss, indent_size + 1) << "          " << value()->toString() << ",\n";
+  indent(ss, indent_size + 1)
+      << "          dropout_p = " << dropout_p() << ",\n";
+  indent(ss, indent_size + 1) << "          is_causal = " << is_causal();
+  if (scale() != nullptr) {
+    indent(ss, indent_size + 1) << ",\n          scale = " << scale();
+  }
```

Comment:
```suggestion
  indent(ss, indent_size + 1)
      << "          dropout_p = " << dropout_p()->toInlineString() << ",\n";
  indent(ss, indent_size + 1) << "          is_causal = " << is_causal()->toInlineString();
  if (scale() != nullptr) {
    indent(ss, indent_size + 1) << ",\n          scale = " << scale()->toInlineString();
  }
```
It's only safe to use `toInlineString()` here if these vals cannot be `TensorView`. It seems like they should always be scalar to me. Can you confirm? If so, should we have some code in the ctor that checks that I couldn't for example pass a `TensorView` for `is_causal`?